packages/transformers/src/tests/idiomatic-test.ts

```diff
@@ -0,0 +1,657 @@
+  let formattedActual;
+  try {
+    formattedActual = prettier.format(unformattedActual, {
+      parser: 'typescript',
+    });
+  } catch {
+    // We might emit invalid TypeScript in a failing test. Rather than fail with
+    // a Prettier parse exception, it's more useful to see a diff.
+    formattedExpected = expectedJs;
+    formattedActual = unformattedActual;
+  }
+  assert.is(formattedActual, formattedExpected);
+  assert.equal(
+    result.diagnostics.map((diagnostic) =>
+      ts.formatDiagnostic(diagnostic, result.host)
+    ),
+    []
+  );
+}
+
+test('@customElement', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {customElement} from 'lit/decorators.js';
+  @customElement('my-element')
+  class MyElement extends LitElement {
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+  }
+  customElements.define('my-element', MyElement);
+  `;
+  checkTransform(input, expected);
+});
+
+test('@property', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {property} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @property()
+    str = "foo";
+
+    @property({type: Number, attribute: false})
+    num = 42;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    constructor() {
+      super(...arguments);
+      this.str = "foo";
+      this.num = 42;
+    }
+    static get properties() {
+      return {
+        str: {},
+        num: {type: Number, attribute: false},
+      };
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@property (merge with existing static properties)', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {property} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    static get properties() {
+      return {
+        str: {},
+      };
+    }
+
+    @property({type: Number})
+    num = 42;
+
+    constructor() {
+      super();
+    }
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    constructor() {
+      super();
+      this.num = 42;
+    }
+    static get properties() {
+      return {
+        str: {},
+        num: {type: Number},
+      };
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@state', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {state} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @state()
+    num = 42;
+
+    @state({hasChanged: () => false})
+    num2 = 24;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    constructor() {
+      super(...arguments);
+      this.num = 42;
+      this.num2 = 24;
+    }
+    static get properties() {
+      return {
+        num: {state: true, attribute: false},
```

Comment:
`state: true` implies `attribute: false` already.

Reply:
Oh, that'd be good too.

Reply:
Sent separately in https://github.com/lit/lit/pull/2030